core/src/main/java/io/grpc/internal/SerializingExecutor.java

```diff
@@ -82,90 +69,48 @@ public SerializingExecutor(Executor executor) {
     this.executor = executor;
   }
 
-  private final Object internalLock = new Object() {
-    @Override public String toString() {
-      return "SerializingExecutor lock: " + super.toString();
-    }
-  };
-
   /**
    * Runs the given runnable strictly after all Runnables that were submitted
    * before it, and using the {@code executor} passed to the constructor.     .
    */
   @Override
   public void execute(Runnable r) {
-    Preconditions.checkNotNull(r, "'r' must not be null.");
-    boolean scheduleTaskRunner = false;
-    synchronized (internalLock) {
-      waitQueue.add(r);
+    runQueue.offer(checkNotNull(r, "'r' must not be null."));
```

Comment:
`assert success;` works fine too. While `offer()` is analogous, there is no implicit pairing between them; use whichever version you need. For instance, when using `BlockingQueue` it is quite normal to mix blocking and non-blocking operations, even though they are in a [similar table](http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/BlockingQueue.html).